hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sources.java

```diff
@@ -92,38 +96,106 @@ private Sources() {
      */
-    public static <K, V> Source<Map.Entry<K, V>> readMap(String mapName) {
+    public static <K, V> Source<Map.Entry<K, V>> readMap(@Nonnull String mapName) {
         return fromProcessor("readMap(" + mapName + ')', SourceProcessors.readMapP(mapName));
     }
 
     /**
      * Returns a source that fetches entries from the Hazelcast {@code IMap}
      * with the specified name, filters them using the supplied predicate,
      * transforms them using the supplied projection function, and emits
-     * the results. Its processors will leverage data locality by fetching
-     * only those entries that are stored on the member where they are running.
+     * them as {@code Map.Entry}. Its processors will leverage data locality
+     * by fetching only those entries that are stored on the member where they
+     * are running.
+     * <p>
+     * The number of Hazelcast partitions should be configured to at least
+     * {@code localParallelism * clusterSize}, otherwise some processors will
+     * have no partitions assigned to them.
+     * <p>
+     * This source does not save any state to snapshot. If the job is restarted,
+     * all entries will be emitted again.
      * <p>
      * If the {@code IMap} is modified while being read, or if there is a
      * cluster topology change (triggering data migration), the source may
      * miss and/or duplicate some entries.
      */
-    public static <K, V, T> Source<T> readMap(String mapName,
-                                              DistributedPredicate<Map.Entry<K, V>> predicate,
-                                              DistributedFunction<Map.Entry<K, V>, T> projectionFn) {
+    public static <K, V, T> Source<T> readMap(@Nonnull String mapName,
+                                              @Nonnull DistributedPredicate<Map.Entry<K, V>> predicate,
+                                              @Nonnull DistributedFunction<Map.Entry<K, V>, T> projectionFn) {
         return fromProcessor("readMap(" + mapName + ')',
                 SourceProcessors.readMapP(mapName, predicate, projectionFn));
     }
 
     /**
+     * Convenience for {@link #streamMap(String, DistributedPredicate,
+     * DistributedFunction, boolean)} with no projection or filtering. It
+     * emits {@link EventJournalMapEvent}s.
+     */
+    @Nonnull
+    public static <K, V> Source<EventJournalMapEvent<K, V>> streamMap(
+            @Nonnull String mapName, boolean startFromLatestSequence) {
+        return fromProcessor("streamMap(" + mapName + ')',
+                SourceProcessors.streamMapP(mapName, startFromLatestSequence));
+    }
+
+    /**
+     * Returns a source that will stream the {@link EventJournalMapEvent}
+     * events of the Hazelcast {@code IMap} with the specified name. Given
+     * predicate and projection will be applied to the events at the source.
+     * <p>
+     * The processors will only access data local to the member and, if {@code
```

Comment:
This is too low-level. We can retain the phrase as in `readMap`, which talks about data locality.